packages/common/src/contexts/accounts.tsx

```diff
@@ -314,23 +314,21 @@ export const getCachedAccount = (
 
 const UseNativeAccount = () => {
   const connection = useConnection();
-  const { wallet } = useWallet();
+  const { publicKey } = useWallet();
 
   const [nativeAccount, setNativeAccount] = useState<AccountInfo<Buffer>>();
 
   const updateCache = useCallback(
     account => {
-      if (wallet && wallet.publicKey) {
-        const wrapped = wrapNativeAccount(wallet.publicKey, account);
-        if (wrapped !== undefined && wallet) {
-          const id = wallet.publicKey?.toBase58();
-          cache.registerParser(id, TokenAccountParser);
-          genericCache.set(id, wrapped as TokenAccount);
-          cache.emitter.raiseCacheUpdated(id, false, TokenAccountParser);
-        }
+      if (publicKey) {
+        const wrapped = wrapNativeAccount(publicKey, account);
+        const id = publicKey.toBase58();
+        cache.registerParser(id, TokenAccountParser);
+        genericCache.set(id, wrapped as TokenAccount);
+        cache.emitter.raiseCacheUpdated(id, false, TokenAccountParser);
       }
     },
-    [wallet],
+    [publicKey],
```

Comment:
This dependency won't work if the key instance changes for the same key. The keys need to be compared using string representation `.toBase58()`.
I know it was like that before and this is not related to this change but wanted to point it out.

Reply:
I don't understand -- the `PublicKey` instance is created when the `WalletAdapter` is connected, and passed through the React context. It should never change unless the wallet reconnects or changes accounts (not implemented yet), and so it should be safe to memoize against.

Reply:
If the instance stays the same then it's ok, but if it does for example when reconnecting do we want to revaluate it?

Reply:
If the adapter disconnects, it will first be set to `null` (in the adapter and in the context) before being set to a `PublicKey`, so on reconnecting even if it was the same instance (which I don't think it can be currently because all the adapters do `new PublicKey(bytes)` on connecting), it should trigger this to be updated anyway since the last value was `null`.